src/ray/gcs/redis_context.cc

```diff
@@ -620,6 +644,14 @@ Status RedisContext::Connect(const std::string &address,
   }
 }
 
+Status RedisContext::Reconnect(RedisConnectType type) {
+  auto type_str = type == CONNECT_SYNC_ONLY
```

Comment:
write a "enum to name" function and call it in the log.

std::string_view MyName(MyType typ);

Reply:
Needs to wait for reply of https://github.com/ray-project/ray/pull/48781#discussion_r1855918089 to see if I need to separate the reconnections of sync context and async context before change this.